hydroflow_plus_test/src/cluster/paxos.rs

```diff
@@ -749,11 +766,12 @@ fn acceptor_p2<'a, P: PaxosPayload, R>(
 
 fn p_p2b<'a, P: PaxosPayload>(
     proposer_tick: &Tick<Cluster<'a, Proposer>>,
-    a_to_proposers_p2b: Stream<P2b<P>, Cluster<'a, Proposer>, Unbounded>,
+    a_to_proposers_p2b: Stream<P2b<P>, Cluster<'a, Proposer>, Unbounded, NoOrder>,
     f: usize,
-) -> Stream<(usize, Option<P>), Cluster<'a, Proposer>, Unbounded> {
+) -> Stream<(usize, Option<P>), Cluster<'a, Proposer>, Unbounded, NoOrder> {
     let (p_broadcasted_p2b_slots_complete_cycle, p_broadcasted_p2b_slots) = proposer_tick.cycle();
-    let (p_persisted_p2bs_complete_cycle, p_persisted_p2bs) = proposer_tick.cycle();
+    let (p_persisted_p2bs_complete_cycle, p_persisted_p2bs) =
+        proposer_tick.cycle::<Stream<_, _, _, NoOrder>>();
     let p_p2b = a_to_proposers_p2b
         .tick_batch(proposer_tick)
         .chain(p_persisted_p2bs);
```

Comment:
Comment for line 780, which I missed before: We still need to compare `p2b.max_ballot` with the proposer's current ballot, not the `p2b.ballot`. Otherwise the acceptor could send p2bs from previous rounds that are incorrectly counted.

Reply:
Hmm how about this scenario though:
1. proposer 1 sends item with ballot 1 to acceptors
2. quorum of acceptors accept item with ballot 1
3. proposer 2 becomes leader due to a heartbeat timeout or something, collects logs from acceptors which includes the items with ballot 1
4. proposer 1 receives quorum of votes from acceptors
5. under your logic, proposer 1 would drop the value, but proposer 2 would pick a slot *after* the item with ballot 1?

Reply:
What do you mean by picking the slot after the item with ballot 1?

Reply:
Like proposer 2 will be aware of the value accepted by the acceptors but dropped by proposer 1, so why does proposer 1 need to drop it?

Reply:
Oh it doesn't matter if proposer 1 drops it. I'm trying to prevent this case:
1. Proposer 1 sends item 1 with ballot 1 to acceptor 1
2. Proposer 2 becomes leader, then proposer 3, then proposer 4, each sending a different item to a different acceptor
3. Proposer 1 becomes leader, send item 1 to acceptor 2, receives ack from acceptor 1 from earlier, believes item 1 is committed
4. Proposer 5 becomes leader and commits a different item

Reply:
AH makes sense, will fix.

Reply:
Wait no actually, since we are doing a `fold_keyed` where the key includes the ballot, won't we be fine because we'll fail to get quorum on item 1 **with** ballot 1 (the item 1 to acceptor 2 will have a different ballot)?